hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -509,22 +509,44 @@ class TestMORDataSource extends HoodieClientTestBase {
 
   @Test
   def testPreCombineFiledForReadMOR(): Unit = {
-    writeData((1, "a0",10, 100))
-    checkAnswer((1, "a0",10, 100))
+    writeData((1, "a0",10, 100, false))
+    checkAnswer((1, "a0",10, 100, false))
 
-    writeData((1, "a0", 12, 99))
+    println("\n\n 111")
+    writeData((1, "a0", 12, 99, false))
     // The value has not update, because the version 99 < 100
-    checkAnswer((1, "a0",10, 100))
+    checkAnswer((1, "a0",10, 100, false))
 
-    writeData((1, "a0", 12, 101))
+    println("\n\n 222")
+    writeData((1, "a0", 12, 101, false))
     // The value has update
-    checkAnswer((1, "a0", 12, 101))
+    checkAnswer((1, "a0", 12, 101, false))
+
+    println("\n\n 333")
+    writeData((1, "a0", 14, 98, false))
+    // Latest value should be ignored if preCombine honors ordering
+    checkAnswer((1, "a0", 12, 101, false))
+
+    println("\n\n 4444")
+    writeData((1, "a0", 16, 97, true))
+    // Ordering value will not be honored for a delete record as the payload is sent as empty payload
+    checkAnswer((1, "a0", 16, 97, true))
+
+    println("\n\n 555")
+    writeData((1, "a0", 18, 96, false))
+    // once a record is deleted, preCombine does not kick. So, any new record will be considered valid ignoring ordering val
+    checkAnswer((1, "a0", 16, 96, false))
```

Comment:
@n3nash : Wanted to discuss this behavior as to whats expected here. I know this is a contrived one, but anyways. Gist is, we first merge all records in the log and then merge with base file and this might run into issues.

format: key, recordVal, preCombine val.
batch1: a0, 1, 100
Result:
snapshot read: a0, 1, 100 // from base file

batch2: a0, 2, 101
Result:
snapshot read: a0, 2, 101 // from merged log file

batch3: a0, 3, 99
Result:
snapshot read: a0, 2, 101 // from merged log file. latest entry will be ignored due to ordering val

batch4: a0, 4, 98 : delete.
Result:
snapshot read: deleted. no record found. 
// Even though ordering val is lower, deletes are sent as empty record payload and hence deletes take precedence. 

batch5: a0, 5, 97 
Result:
snapshot read: 
expected: once a record is deleted, any new insert will be considered valid ignoring honoring val. hence this should return a0,5,97 ignoring all previous values (for ordering val)
actual: a0, 1, 100 // record from base file. 
Guess we first merge all records in log files which results in a0,5,97 (active) and then combine with base file which has a0,1,100 and hence the results is a0,1,100